packages/scope-manager/tests/fixtures/type-declaration/type-query-qualified.ts

```diff
@@ -0,0 +1,3 @@
+const x = { y: { z: 1 } };
+type T = typeof x.y.z;
+type Unresolved = x.y.z;
```

Comment:
those test can/should be created added separatelly as they validate execute previously uncovered branches